src/modules/user/repository/user.repository.ts

```diff
@@ -0,0 +1,46 @@
+import { ClassProvider, Injectable } from "@nestjs/common";
+import {
+	RepoCreateUserInput,
+	RepoUser,
+	UserRepositoryDefinition,
+	UserRepositoryName,
+} from "src/modules/user/repository/user.repository.types";
+import { DataSource } from "typeorm";
+import {UserRecord} from "src/core/infrastructure/entities/User";
+
+@Injectable()
+export class UserRepository implements UserRepositoryDefinition {
+	constructor(private dataSource: DataSource) {}
```

Comment:
DataSource is just the database connection. The module defines what we're allowed to access. See `user.module.ts`